### ISSUE TITLE
Unexpected geo variables result in unexpected rows for certain geo hierarchies

### ISSUE THREAD

**Issue by cmsetzer** [socrata/autocensus#11]
As an example, the following query results in extra rows where `variable_code` is `county`:

```python
from autocensus import Query
​
query = Query(
    estimate=5,
    years=range(2017, 2019),
    variables=['S0101_C01_028E'],
    for_geo='tract:*',
    in_geo=['state:34'],
    join_geography=True
)
data = query.run()
print(data.shape)
print(data['variable_code'].value_counts())
data[data['variable_code']=='county'].head()
```

In [this associated API call](https://api.census.gov/data/2018/acs/acs5/subject?get=NAME,S0101_C01_001E&for=tract:*&in=state:34), the intermediate geography `county` is being returned even though not explicitly requested. We need to add some logic to either filter out these unexpected geography variables or pivot them along with the expected ones.

Thanks to @aliciatb for catching this!

**Comment by aliciatb** [comment]
When county is passed `in_geo`, the code works fine!

` in_geo=['state:34', 'county:*'],`

**Comment by cmsetzer** [comment]
This came up again with another query so going to reopen for future troubleshooting.

Here's a query that resulted in weird extra rows for a user:

```
import os

import pandas as pd

from autocensus import Query
from autocensus.socrata import to_socrata

# High Risk Indicators and Demographics
pop_60_var = "S0101_C01_028E"
residents_no_health_insurance_by_gender_age = [
    "B27001_005E",
    "B27001_008E",
    "B27001_011E",
    "B27001_014E",
    "B27001_017E",
    "B27001_020E",
    "B27001_023E",
    "B27001_026E",
    "B27001_029E",
    "B27001_033E",
    "B27001_036E",
    "B27001_039E",
    "B27001_042E",
    "B27001_048E",
    "B27001_051E",
    "B27001_054E",
    "B27001_057E",
]
poverty_var = "B17015_002E"
birth_var = "B13002_002E"
language_vars = [
    "C16001_002E",
    "C16001_003E",
    "C16001_006E",
    "C16001_009E",
    "C16001_012E",
    "C16001_015E",
    "C16001_018E",
    "C16001_021E",
    "C16001_024E",
    "C16001_027E",
    "C16001_030E",
    "C16001_033E",
    "C16001_036E",
]
no_internet_var = "B28002_013E"
snap_var = "S2201_C03_001E"
pop_var = "B01003_001E"

# customer specific settings
custom_name = "State of Missouri Census Tracts"  # appended to dataset name
in_geo = "state:29"
domain = "setzer-scgc.demo.socrata.com"

# user spcific settings
username = os.environ["SOCRATA_USERNAME"]
password = os.environ["SOCRATA_PASSWORD"]

# 1/5 year - for Tract level and populations < 65K need to use 5 year estimates
acs_year_estimate = 5

all = []
all.append(pop_60_var)
all.append(poverty_var)
all.append(birth_var)
all.append(snap_var)
all.append(no_internet_var)
all.append(pop_var)
for i in residents_no_health_insurance_by_gender_age:
    all.append(i)
for i in language_vars:
    all.append(i)

# tract level by State and County
query = Query(
    estimate=5,
    years=range(2018, 2019),
    variables=all,
    for_geo="tract:*",
    in_geo=in_geo,
    join_geography=True,
)
data = query.run()
```

**Comment by cmsetzer** [comment]
I believe https://github.com/socrata/autocensus/commit/bcdf450d5c95da6d57e8461b7c5138dea8b82c98 fixes this. It seems that the Census API returns fields representing intermediate levels of geo hierarchy even if you don't explicitly request them (e.g., counties if you request `for=tract:*&in=state:29`). We needed to properly exclude all extra geo fields from our unpivoted output.

This fix is merged in the development branch and will go out in the next release. Closing this issue again, thanks again @aliciatb for first spotting this!
